transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -569,21 +580,39 @@ def _process_scan_event(self, response):
 
             if data[22] == 0xFF and data[23] == 0xC0 and data[24] == 0x3:
                 self._v1_scan_count += 1
-                self._parse_v1_advertisement(rssi, string_address, data)
+                info = self._parse_v1_advertisement(rssi, string_address, data)
             elif data[3] == 27 and data[4] == 0x16 and data[5] == 0xdd and data[6] == 0xfd:
                 self._v2_scan_count += 1
-                self._parse_v2_advertisement(rssi, string_address, data)
+                info, reading_time, stream, reading, \
+                    broadcast_toggle, counter, broadcast_multiplex = \
+                    self._parse_v2_advertisement(rssi, string_address, data)
             else:
                 pass # This just means the advertisement was from a non-IOTile device
         elif packet_type == 4:
             self._v1_scan_response_count += 1
-            self._parse_v1_scan_response(string_address, data)
+            info, reading_time, stream, reading = \
+                self._parse_v1_scan_response(string_address, data)
+
+        if info:
+            if self._throttle_broadcast and self._check_update_seen_broadcast(
+                        sender, reading_time, stream, reading,
+                        broadcast_toggle, counter=counter, channel=broadcast_multiplex):
+                return
+
+            self._trigger_callback('on_scan', self.id, info, self.ExpirationTime)
```

Comment:
As discussed on the call, we should add a second config variables (`throttle_scans`) that controls whether these scans are throttled independently of the broadcast values.

Reply:
Yes, I added throttle-scan and throttle-scan-timeout to enable disable filtering of device_seen events and configure the timeout after which a same report or scan will be repeated